### PR TITLE
Add required  docs for terraform store

### PR DESCRIPTION
**Pull request by drewatx**
### What does this PR do?
- This PR adds the docs needed for this provider to be published to the terraform store. I created the `docs` folder at the root of the repo which contains all documentation for terraform. 

### What ticket does this PR close?
- There is no ticket for this PR to my knowledge (I don't have access to a board tho)

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [ ] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [ ] This PR does not require updating any documentation

### CODE REVIEW

File: docs/data-sources/instance.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# <resource name> Resource/Data Source
```
Comment (jtuttle) [inline docs/data-sources/instance.md:1]: Do we need someone from C&I to fill this in?

Comment (drewatx) [reply docs/data-sources/instance.md:1]: @jtuttle This is still in progress, I'm working on filling this in to the best I can with my limited knowledge of this integration. Maybe once we get this published to the store someone who is more knowledgeable can come back and update these.

Comment (jtuttle) [reply docs/data-sources/instance.md:1]: Ah okay thanks. I'm just going to change the name to include `WIP`, that's our usual method for indicating incomplete PRs.

File: docs/index.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
+# <provider> Provider
```
Comment (ismarc) [inline docs/index.md:1]:
```suggestion
# Conjur Provider
```

**Review by ismarc — CHANGES_REQUESTED** [review]

**Review by ismarc — APPROVED** [review]

**Review by ismarc — APPROVED** [review]

**Pull request merged**